### PR TITLE
Pipes connection based on rotation

### PR DESCRIPTION
**Pull request by Aranclanos**
### Purpose
Vents, scrubbers, connectors and shuttleheaters will only connect from one side.
Pipes will now connect eachother based on their direction instead of always north/south.

### Open Questions and Pre-Merge TODOs

- [X]  This fix is tested on the same branch it is PR'ed to.
- []  I correctly commented my code
- [X]  My code is indented with tabs and not spaces
- [X]  This PR does not include any unnecessary .meta, .prefab or .unity (scene) changes
- [X]  This PR does not bring up any new compile errors
- [X]  This PR has been tested in editor
- [ ]  This PR has been tested in multiplayer (with 2 clients and late joiners, if applicable)

### Notes:
later on all the direction code (including cables) should be merged into one

### CODE REVIEW

File: UnityProject/Assets/Scripts/Shuttles/ShuttleHeater.cs [hunk 1]
```diff
@@ -19,4 +19,32 @@ public override void SetAnchored(bool value)
...
+
+	public override void CalculateDirection()
+	{
+		direction = 0;
+		var rotation = transform.rotation.eulerAngles.z;
+		if (rotation >= 45 && rotation <= 135)
+		{
+			direction = Direction.WEST;
+		}
+		else
```
Comment (Jibbajabbafic) [inline UnityProject/Assets/Scripts/Shuttles/ShuttleHeater.cs:31]: Why not use `else if (...)` instead?

File: UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/Pipenets/Pipe.cs [hunk 2]
```diff
@@ -17,13 +18,16 @@ public class Pipe : MonoBehaviour
...
 	{
-		NORTH,
-		SOUTH,
-		WEST,
-		EAST
+		NONE = 0,
+		NORTH = 1,
+		SOUTH = 2,
+		WEST = 4,
+		EAST = 8
```
Comment (Jibbajabbafic) [inline UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/Pipenets/Pipe.cs:28]: Good way of doing the values
```suggestion
		NORTH = 1 << 0,
		SOUTH = 1 << 1,
		WEST  = 1 << 2,
		EAST  = 1 << 3
```

File: UnityProject/Assets/Scripts/Shuttles/ShuttleHeater.cs [hunk 3]
```diff
@@ -19,4 +19,32 @@ public override void SetAnchored(bool value)
 	{
 		anchored = value;
 	}
+
+	public override void CalculateDirection()
+	{
+		direction = 0;
+		var rotation = transform.rotation.eulerAngles.z;
+		if (rotation >= 45 && rotation <= 135)
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/Shuttles/ShuttleHeater.cs:27]: All of this reminds me of something...
https://github.com/unitystation/unitystation/blob/5aa2bc64d6a191068f618000c86ef218ae11b5c7/UnityProject/Assets/Scripts/Transform/Orientation.cs#L147-L160
Why not use Orientation for this?

Comment (fomalsd) [reply UnityProject/Assets/Scripts/Shuttles/ShuttleHeater.cs:27]: 'cause, you know, you're against code duplication and still used `if (rotation >= 45) {blablabla}` thingy in three places within this PR
I mean I can merge this now, but are you planning to refactor this later?

Comment (Aranclanos) [reply UnityProject/Assets/Scripts/Shuttles/ShuttleHeater.cs:27]: thats the plan

**Pull request merged**